features/step_definitions/journey_test_steps.rb

```diff
@@ -47,6 +68,10 @@
   click_link_or_button(button_name)
 end
 
+When /I click the '(.*)' button for the supplier user '(.*)'$/ do |button_name,user_name|
+  find(:xpath, "//*/span[contains(text(),'#{user_name}')]/../../td/*//button[contains(text(),'#{button_name}')]").click
```

Comment:
Yeah, I'm going to have to do something similar.   
Basically, I need to check a user that's been removed by a supplier has an 'activate' (not a 'deactivate' button) in its row.